uiowa/uiowa
# Limit number of upcoming dates displayed on the MAUI block

Resolves #5072 

# Story
(Taken from the linked issue): As an editor using the MAUI upcoming dates feature, I would like to limit the number of items displayed, so that I can place a limited number of items on the homepage of my site and link to another page or the Registrar's website for additional upcoming dates.

# How to test
1. Sync down a site. 
2. If the MAUI module isn't already enabled (aka you're not syncing down Grad,) `ddev drush @yoursite.local en uiowa_maui`. Then, on `/admin/structure/types/manage/page/display`, make sure 'Allow all existing & new MAUI blocks', found under 'Blocks available for placement (all regions & placements)', is selected.
4. Add a MAUI block somewhere, test that the new options for limiting the number of deadlines displayed works as intended. 
5. A lot of the description text and default values are set up to match the articles and events listblocks, but some is custom. Check it makes sense.
6. Check articles and events listblocks still work as intended, as a function used by them was moved out of listblock.php and to its own file, so that it could also be called by the MAUI block.

Bonus points: sync down grad and make sure this doesn't mess up any existing blocks, such as on https://grad.uiowa.edu/academics/deadlines
